window-purpose-x.el

```diff
@@ -653,13 +653,22 @@ This function removes the buffer denoted by BUFFER-OR-NAME from all
 window-local buffer lists."
   (interactive "bBuffer to replace: ")
   (let* ((buffer (window-normalize-buffer buffer-or-name))
-         (purpose (purpose-buffer-purpose buffer))
-         (other-buffers (delete buffer (purpose-buffers-with-purpose purpose))))
+         ;; Delay calculating other-buffers until we need it
+         ;; This prevents unnecessary calculations on temporary
+         ;; buffers created by `with-temp-buffer' and other likewise
+         ;; non-displayed buffers
+         (have-other-buffers nil)
+         (other-buffers nil))
```

Comment:
Looks good, but the name `have-other-buffers` is confusing. How about `other-buffers-calculated`?

Reply:
Got it